spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Log.java

```diff
@@ -40,7 +43,12 @@
 	 */
 	private final String event;
 
-	public Log(long timestamp, String event) {
+	@JsonCreator
+	public Log(
+			@JsonProperty(value = "timestamp", required = true) long timestamp,
```

Comment:
if you use a constructor as opposed to reflection, it is. The "required"
flag isn't .. uhh.. required. I can switch to the reflective annotations,
if you like.